### PR TITLE
Series of small WCOSS2 updates

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
Updates included in this PR:

1. WCOSS2.env updates for CFP
2. a couple small waveinit job script updates (runs without failure now)
3. build updates for enkf_chgres_recenter_nc (including conversion to LUA modulefile)
4. change NWROOT to PACKAGEROOT in exglobal_forecast.sh
5. update GSI and WAFS component checkouts

Refs: #399

### CODE REVIEW

File: sorc/enkf_chgres_recenter_nc.fd/makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 SHELL=  /bin/sh
 
-LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(NETCDF)/lib -lnetcdff -lnetcdf -lhdf5_hl -lhdf5 -lz
+LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(NETCDF)/lib -L$(HDF5_LIBRARIES) -lnetcdff -lnetcdf -lhdf5_hl -lhdf5 -lz
```
Comment (aerorahul) [inline sorc/enkf_chgres_recenter_nc.fd/makefile]:
```suggestion
LIBS= $(FV3GFS_NCIO_LIB) $(BACIO_LIB4) $(W3NCO_LIB4) $(IP_LIB4) $(SP_LIB4) -L$(HDF5_LIBRARIES) -lhdf5_hl -lhdf5 -lz -L$(NETCDF)/lib  -lnetcdff -lnetcdf 
```

**Review by aerorahul — COMMENTED** [review]
looks good. 
Just one comment.

**Comment by KateFriedman-NOAA** [comment]
@aerorahul I have also included two additional small commits to this PR, which resolve a failing waveprep job (similar to waveinit updates in this PR). Please review and approve if good, thanks!

**Review by aerorahul — APPROVED** [review]

File: sorc/checkout.sh [hunk 2]
```diff
@@ -35,9 +35,9 @@ fi
 echo gsi checkout ...
 if [[ ! -d gsi.fd ]] ; then
     rm -f ${topdir}/checkout-gsi.log
-    git clone --recursive https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
+    git clone --recursive --branch feature/gfsda.v16.1.5_wcoss2_port https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
     cd gsi.fd
-    git checkout gfsda.v16.1.1a
+    #git checkout feature/gfsda.v16.1.5_wcoss2_port
     git submodule update
```
Comment (WalterKolczynski-NOAA) [inline sorc/checkout.sh:41]: Should this be recursive, just in case another layer of submodules is added?

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: sorc/checkout.sh (hunk 2 above)
Comment (KateFriedman-NOAA) [reply sorc/checkout.sh:41]: On line 38 of checkout.sh I am recursively cloning the branch and commented out the secondary branch checkout on line 40 for now.

```
git clone --recursive --branch feature/gfsda.v16.1.5_wcoss2_port https://github.com/NOAA-EMC/GSI.git gsi.fd >> ${topdir}/checkout-gsi.log 2>&1
```

Comment (WalterKolczynski-NOAA) [reply sorc/checkout.sh:41]: That works too. You shouldn't even need to do the submodule update that way.

We should probably pick one way and do all components the same way. Specifying the branch at clone time is probably best.

Comment (KateFriedman-NOAA) [reply sorc/checkout.sh:41]: Agreed. The checkout script is a little messy right now as I set the test branches but I'd like to review and clean it towards the end as we get tags sorted.

**Pull request merged**